knimin/lib/vioscreen.py

```diff
@@ -0,0 +1,446 @@
+import requests
+import os
+import functools
+import json
+import pandas as pd
+
+from datetime import datetime
+from knimin import config
+from knimin.lib.data_access import SQLHandler
+
+class VioscreenHandler(object):
+    def __init__(self):
+        with open('./auth.json') as f:
+            self._key = json.load(f)['key']
+            self._user = json.load(f)['user']
+            self._pw = json.load(f)['pw']
+        self._session = requests.Session()
+        # setup our HTTP header data
+        self._headers = None
+        self._users = None
+        self.sql_handler = SQLHandler(config)
+
+    # get an API token
+    def get_token(self):
+        """Gets an API token for vioscreen
+
+        Return
+        ------
+        str
+            The API token
+        """
+        return self.post('https://api.viocare.com/%s/auth/login' % self._key,
+                     data={"username": self._user, 
+                           "password": self._pw})['token']
+    
+    def get_users(self):
+        """Gets list of users that vioscreen has data for
+
+        Return
+        ------
+        list of dict
+            List of users that have vioscreen data
+        """
+        if not self._headers:
```

Comment:
would it make sense for `self._headers` to get set in the constructor?

Reply:
Setting the headers takes the most time. I remember showing you running the script and it taking some time, so I made this change so testing would be expedient if I didn't need the headers. But I can see why I could just set it in the constructor. Would you still want me to go through with the change.

Reply:
There's a boilerplate suggestion below too, which could probably be minimally extended to use as a decorator

Reply:
Hm I just moved the headers and users initialization back to the constructor, which would hypothetically have headers and users set always. Would this suffice?